src/screens/Departures/state/stop-place-state.ts

```diff
@@ -278,15 +286,15 @@ export function useStopPlaceData(
       }),
     [stopPlace.id, favoriteDepartures, showOnlyFavorites],
   );
-  useEffect(refresh, [stopPlace.id, startTime]);
+  useEffect(dispatchLoadInitialDepartures, [stopPlace.id, startTime]);
```

Comment:
Why are we not using refresh here? Are we not losing the load departures being wrapped in a callback hook because of this?

Reply:
I am not the best at using react! 😅. But, I have noticed that `refresh` is a `useCallback` implementation with its own dependencies and is called inside a `useEffect` that also contains its own dependencies and looks a bit odd, according to the documentation https://reactjs.org/docs/hooks-reference.html#usecallback the use of `useCallback` are mainly used to be returned by the child view to trigger updates, so no make sense double call it inside the same child. So because `refresh` is still returned to the parent the hook implementation is not being lost, I have tried and have not seen anything failed or strange but I might be wrong. (Long explanation)

Reply:
I think it would be fine here as refresh was anyway not being called if anything other than stopPlace.id or startTime is changing, but probably at the other place you still want to keep refresh. SO its wrapped in a callback and return memoised value unless the refresh dependencies are changed.

Reply:
there should be a failed result as a consequence, but only calling expensive functions when they dont need to be I think :)